main/lib/server.js

```diff
@@ -268,15 +268,19 @@ export class HttpServer {
    * @param {HttpWTServerSessionVisitorEvent} args
    */
   onHttpWTSessionVisitor(args) {
+    // match session streams on path without query string
+    const path = args.path.split('?')[0]
```

Comment:
As this is not the header path field, we should not temper here. So please remove.

Reply:
The `args` passed here during the new `should connect to a path with a query in the URL` test are:

```js
{
  session: Http3WTSessionVisitor {},
  path: '/session_with_query?foo=bar'
}
```

If we don't remove the query string from the path, the session is never passed to the session stream controller for the endpoint and the test fails.